### PR TITLE
Filtering of symptoms now works in the eating section

### PR DESCRIPTION
**Pull request by brettminnie**

### CODE REVIEW

File: src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/SymptomGenerator.php [hunk 1]
```diff
@@ -46,8 +68,7 @@ protected function extractCause(array $symptoms)
             $slice = array_shift($symptoms);
         } while(count($slice) === 0);
 
-        return $slice[(int)rand(0, count($slice)-1)];
-
+        return $slice[(int)rand(0, count($slice) - 1)];
```
Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/SymptomGenerator.php]: May be a comment here?

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/SymptomGenerator.php]: Have to document the whole thing to be fair

But can do

Comment (eddiejaoude) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/SymptomGenerator.php]: Just a single line would be useful - why `rand` & `-1`

Comment (brettminnie) [inline src/TransformCore/PHE/HayApi/NarrativeBundle/Narrative/SymptomGenerator.php]: Surely I don't need to explain why count(array)-1 is used as a maximum
array bound

The rand sure

**Pull request merged**